### ISSUE TITLE
Microfone sound rattles and cuts

### ISSUE THREAD

**Issue by vscfreire** [FreeRDP/FreeRDP#6835]
**Description**
The microphone sound slightly rattles and cuts with PCM from Windows client to Windows server. If we try to use AAC_MS (0xa106) the rattles and cuts are very prevalent. From a Linux client, PCM sounds perfect but AAC_MS rattles, although much less that in Windows.

**To Reproduce**
1. Connect to a Windows Server (2016) with a FreeRDP stable-next release, with mic on.
2. Use https://webcammictest.com/check-mic.html to record at least 20 seconds of continuous mic sound.
3. Playback the recorded sound.

**Expected behavior**
No sound rattles from mic recording.

**Application details**
* FreeRDP version: stable-next
* Command line used: /mic
* FreeRDP compiled with FFmpeg 4.2.4 and DSP Experimental for AAC_MS
* OS version connecting to (server side): Windows Server 2016

**Environment (please complete the following information):**
 - OS: Windows 10 Version 10.0.19042 Build 19042

**Additional context**
This sound rattle that gets worse with more modern codecs remind me of issue #6490, that has since been fix in sound output.

**Comment by akallabeth** [comment]
@vscfreire the `AAC` related stuff is known (that is the reason you need the `EXPERIMENTAL` compile flag ;)

**Comment by vscfreire** [comment]
> @vscfreire the `AAC` related stuff is known (that is the reason you need the `EXPERIMENTAL` compile flag ;)

Yep, I understand. The AAC is for reference and because the sound experience was similar to #6490. The slightly rattles and cuts with PCM from Windows client to Windows server is the main issue.

**Comment by akallabeth** [comment]
@vscfreire ok, tested with current source, no `ffmpeg` compiled in (just using plain `PCM` data) and the sound is perfectly clear.
Sure that you´re using `PCM` and not some other encoded format for the tests if using `FFMPEG`? (depending on bandwidth that might switch during runtime)

On a note, it is not similar to #6490 as for microphone we do not have to fill a playback buffer (in time), but we get a callback with data (and size) already available and just send that to the server.
Could you share your build configuration, maybe it is related to that (depending on optimization/flags/...)

**Comment by vscfreire** [comment]
Thanks for testing @akallabeth.

I have now reproduced the rattle in Microsoft RDP client. It's the OpenVPN connection to the Windows RDS server network. When I tested from Linux, I was direct, but from Windows, I used the VPN to reach the server. It's a high bandwidth stable connection but the VPN must be introducing some kind of latency/fragmentation that makes the sound recording rattle.

Probably this would be something that the server side RDP could/should be optimized to handle, but it's clearly out of FreeRDP scope. Thank you for your time.
